### ISSUE TITLE
C++ infer plugin failure of TensorRT 8.6.0.1 when running trt inferece on GPU 3090

### ISSUE THREAD

**Issue by QingXIA233** [NVIDIA/TensorRT#3441]
## Description

I have this onnx model, when I use trtexec command to export it to tensorrt engine in FP16 mode with TensorRT , everything works fine, the engine was generated without any trouble. I even checked the outputs of trt and torch models, they are consistent, only some precision loss wrt the fp16 mode.
![image](https://github.com/NVIDIA/TensorRT/assets/64100948/9b983edf-f306-42c9-adff-d146a04abe51)
However, when I try to use this trt model to perform model inference with C++:
![image](https://github.com/NVIDIA/TensorRT/assets/64100948/0f3c8e7a-9a41-49f4-a4a0-bc108cbc4d45)
![image](https://github.com/NVIDIA/TensorRT/assets/64100948/475c14ec-0fb1-4f6e-8cf0-baba811e3bd6)
it gets error, code: 716, reason:misaligned address:
![image](https://github.com/NVIDIA/TensorRT/assets/64100948/88db8358-adcf-4cad-b9d4-9e898ec5bf9b)

I also used cuda-memcheck to figure out the exact error location, and I get this log. if the log file is required, I could provide:
![image](https://github.com/NVIDIA/TensorRT/assets/64100948/663aee6c-3a28-4a7a-bfb3-dcab7b0941f2)
I used cuda-memcheck to figure out the exact error location, and I get this log. if the log file is required, I could provide:
![image](https://github.com/NVIDIA/TensorRT/assets/64100948/b4454cdb-6e0a-4ab5-8878-cff46a710e85)

## Environment

**TensorRT Version**: 8.6.0.1

**NVIDIA GPU**:  3090Ti

**NVIDIA Driver Version**: 515.48.07

**CUDA Version**: 11.4

Operating System: linux

Python Version (if applicable): 3.7

PyTorch Version (if applicable):1.8.0+cu101

## Relevant Files

**Model link**: https://drive.google.com/drive/folders/1cWR-eszD4sUXxEOiVmKeLwbHIIO1XcuO?usp=sharing
**Error log**: https://drive.google.com/file/d/1svH0-PhgMPU-BODgkKvgdz2vJx9xhWcU/view?usp=sharing

## Steps To Reproduce
1. use trtexec to get the trt model (ok)
2. use C++ infer plugin to run trt infer (not ok)

**Commands or scripts**: trtexec --onnx=model_vanilla.onnx --saveEngine=model_vanilla_fp16.trt --fp16 --workspace=10240

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**: Yes

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`):  Yes

```[tasklist]
### Tasks
- [ ] https://github.com/NVIDIA/TensorRT/issues/2776
```

**Comment by QingXIA233** [comment]
FYI, when I use TensorRT8.2, both fp32 and fp16 trt models work totally well. However, when I switched to TensorRT8.6, only fp32 trt model could work, fp16 trt model failed no matter what I tried. The only solution I can think of is forcing several layers to be fp32 mode. But this would undermining my initial intent of using 8.6 which makes my model run faster. Therefore, I really need some advice here. Is the mixed fp32-fp16 precision is essential in my case? Or is there any solution to solving the misaligned address problem?

**Comment by zerollzeng** [comment]
I'm pretty sure that there are bugs in your inference code since trtexec can load and run the engine successfully.

**Comment by zerollzeng** [comment]
Double check the buffer_dev_, it should contains the device buffer pointer associated with you engine.

**Comment by QingXIA233** [comment]
> I'm pretty sure that there are bugs in your inference code since trtexec can load and run the engine successfully.

@zerollzeng Thank you a lot for replying. Yes, I agree with the fact that there is some error in the infer code but I couldn't locate. What really confuses me is that: I use **the same inference code** to run C++ inference for TensorRT 8.2 model and TensorRT 8.6 model (same onnx file), the former runs perfectly well, the latter gets misaligned address.  Is there **any C++ infer plugin that is neccessary** **to be modified** when I use TensorRT 8.6? Could you please give me some hints here?  BTW, I also changed to use enqueueV3 plugin instead of enqueueV2 according to the official TensorRT8.6 documents, but it still failed.

**Comment by zerollzeng** [comment]
I can't give useful suggestion in your case, but I think you can take a look at the trtexec source code and compare the diff between TRT 8.2 and 8.6.

**Comment by ttyio** [comment]
@QingXIA233  do you have any customer plugins, because cuda is async I am not sure if this is delayed from the plugin error.

Also have tried the TRT 9.1 in OSS container? If there is no customer plugin, let's check if this is fixed issue in 9.1, thanks!
https://github.com/NVIDIA/TensorRT/tree/release/9.1
